mint-auditor/src/gnosis/eth_data_types.rs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! Type wrappers for Ethereum addresses.
+
+use super::Error;
+use serde::{Deserialize, Serialize};
+use std::{fmt, str::FromStr};
+
+/// Ethereum 20 byte address.
+/// We currently do not store the decoded bytes since we want to maintain the
+/// original capitalization (which is how Ethereum addresses represent a
+/// checksum). We don't have a need for the raw bytes.
+#[derive(Clone, Debug, Default, Deserialize, Serialize)]
+pub struct EthAddr(pub String);
+
+impl FromStr for EthAddr {
+    type Err = Error;
+
+    fn from_str(src: &str) -> Result<Self, Self::Err> {
+        if !src.starts_with("0x") {
```

Comment:
Since these are case insensitive do we need to worry about someone doing `blah.upper()` and it coming in as `0X1234....`? 
I'm guessing it's probably fine as is

Reply:
Ethereum uses capitalization in the hex digits for checksumming (https://coincodex.com/article/2078/ethereum-address-checksum-explained/)
Technically the `0x` part is expected to be lowercase either way.